src/libs/decoders/SDL_sound.c

```diff
@@ -32,13 +32,13 @@
 #  include <config.h>
 #endif
 
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
-#include <ctype.h>
+#ifdef _MSC_VER
+// Avoid warning about use of strncpy
+#define _CRT_SECURE_NO_WARNINGS
+#endif
```

Comment:
Before you started dosbox-staging, I was conservative in making changes fearing updates to my patch would break builds for users on mingw / VisualStudio / OS X (I did build testing up to some point and then it became too much manual labour). 

But now with CI (covering the majority of the permutations between hosts and compilers), I have almost no fear I'll break people's builds; so I want to make the code as safe and correct as possible. 

I'll try for snprintf and fallback if some compilers don't support it.

Reply:
Fixed in a19570b893e789908e5abc54c48e35b956945a70